File: src/main/java/br/com/correios/enderecador/dao/GpbeCEPFactory.java
package br.com.correios.enderecador.dao;

public class GpbeCEPFactory extends CEPFactory {
    private static GpbeCEPFactory instance;

    public static synchronized GpbeCEPFactory getIntance() {
        if (instance == null)
            instance = new GpbeCEPFactory();
        return instance;
    }

    public InterfaceEnderecoDao getEndereco() {
        return EnderecoGpbeDao.getInstance();
    }
}

File: src/main/kotlin/br/com/correios/enderecador/dao/GpbeCEPFactory.kt
package br.com.correios.enderecador.dao

import kotlin.jvm.Synchronized

class GpbeCEPFactory : CEPFactory() {
    override val endereco: InterfaceEnderecoDao?
        get() = EnderecoGpbeDao.instance

    companion object {
        private var instance: GpbeCEPFactory? = null

        @get:Synchronized
        val intance: GpbeCEPFactory?
            get() {
                if (instance == null) {
                    instance = GpbeCEPFactory()
                }
                return instance
            }
    }
}
